### DISCUSSION TITLE
3.9.0-alpha02

### DISCUSSION THREAD

**Discussion by Him188** [open-ani/animeko#895] [Announcements]
- 修复安卓上的弹幕渲染问题 (自上个版本)
- 安卓自动隐藏控制器
- 修复 Windows 退出全屏时的窗口大小问题

**Full Changelog**: https://github.com/open-ani/ani/compare/v3.9.0-alpha01...v3.9.0-alpha02

----

### 下载

<details>
<summary>>> <h3>点我展开下载列表</h3> <<</summary>

[//]: # (注意, `checkLatestVersion` 有字符串处理, 修改标题和分隔符前务必查询)

[github-win-x64]: https://github.com/open-ani/ani/releases/download/v3.9.0-alpha02/ani-3.9.0-alpha02-windows-x86_64.zip

[github-mac-x64]: https://github.com/open-ani/ani/releases/download/v3.9.0-alpha02/ani-3.9.0-alpha02-macos-x86_64.dmg

[github-mac-aarch64]: https://github.com/open-ani/ani/releases/download/v3.9.0-alpha02/ani-3.9.0-alpha02-macos-aarch64.dmg

[github-android]: https://github.com/open-ani/ani/releases/download/v3.9.0-alpha02/ani-3.9.0-alpha02-universal.apk

[github-android-arm64-v8a]: https://github.com/open-ani/ani/releases/download/v3.9.0-alpha02/ani-3.9.0-alpha02-arm64-v8a.apk

[github-android-armeabi-v7a]: https://github.com/open-ani/ani/releases/download/v3.9.0-alpha02/ani-3.9.0-alpha02-armeabi-v7a.apk

[github-android-x86_64]: https://github.com/open-ani/ani/releases/download/v3.9.0-alpha02/ani-3.9.0-alpha02-x86_64.apk

[cf-win-x64]: https://d.myani.org/v3.9.0-alpha02/ani-3.9.0-alpha02-windows-x86_64.zip

[cf-mac-x64]: https://d.myani.org/v3.9.0-alpha02/ani-3.9.0-alpha02-macos-x86_64.dmg

[cf-mac-aarch64]: https://d.myani.org/v3.9.0-alpha02/ani-3.9.0-alpha02-macos-aarch64.dmg

[cf-android]: https://d.myani.org/v3.9.0-alpha02/ani-3.9.0-alpha02-universal.apk

[cf-android-arm64-v8a]: https://d.myani.org/v3.9.0-alpha02/ani-3.9.0-alpha02-arm64-v8a.apk

[cf-android-armeabi-v7a]: https://d.myani.org/v3.9.0-alpha02/ani-3.9.0-alpha02-armeabi-v7a.apk

[cf-android-x86_64]: https://d.myani.org/v3.9.0-alpha02/ani-3.9.0-alpha02-x86_64.apk

[ghproxy-win-x64]: https://mirror.ghproxy.com/?q=https%3A%2F%2Fgithub.com%2Fopen-ani%2Fani%2Freleases%2Fdownload%2Fv3.9.0-alpha02%2Fani-3.9.0-alpha02-windows-x86_64.zip

[ghproxy-mac-x64]: https://mirror.ghproxy.com/?q=https%3A%2F%2Fgithub.com%2Fopen-ani%2Fani%2Freleases%2Fdownload%2Fv3.9.0-alpha02%2Fani-3.9.0-alpha02-macos-x86_64.dmg

[ghproxy-mac-aarch64]: https://mirror.ghproxy.com/?q=https%3A%2F%2Fgithub.com%2Fopen-ani%2Fani%2Freleases%2Fdownload%2Fv3.9.0-alpha02%2Fani-3.9.0-alpha02-macos-aarch64.dmg

[ghproxy-android]: https://mirror.ghproxy.com/?q=https%3A%2F%2Fgithub.com%2Fopen-ani%2Fani%2Freleases%2Fdownload%2Fv3.9.0-alpha02%2Fani-3.9.0-alpha02-universal.apk

[ghproxy-android-arm64-v8a]: https://mirror.ghproxy.com/?q=https%3A%2F%2Fgithub.com%2Fopen-ani%2Fani%2Freleases%2Fdownload%2Fv3.9.0-alpha02%2Fani-3.9.0-alpha02-arm64-v8a.apk

[ghproxy-android-armeabi-v7a]: https://mirror.ghproxy.com/?q=https%3A%2F%2Fgithub.com%2Fopen-ani%2Fani%2Freleases%2Fdownload%2Fv3.9.0-alpha02%2Fani-3.9.0-alpha02-armeabi-v7a.apk

[ghproxy-android-x86_64]: https://mirror.ghproxy.com/?q=https%3A%2F%2Fgithub.com%2Fopen-ani%2Fani%2Freleases%2Fdownload%2Fv3.9.0-alpha02%2Fani-3.9.0-alpha02-x86_64.apk

[qb-enhanced]: https://github.com/c0re100/qBittorrent-Enhanced-Edition/releases/latest

- PC 首次播放在线数据源时, 可能需要加载 10-30 秒。
- macOS
安装后如果无法打开: [解决方案](https://github.com/open-ani/ani/wiki/macOS-%E6%97%A0%E6%B3%95%E6%89%93%E5%BC%80%E8%A7%A3%E5%86%B3%E6%96%B9%E6%A1%88)
- Windows
打开时如果显示异常: [解决方案](https://github.com/open-ani/ani/wiki/Windows-%E4%B8%8B%E5%AD%97%E4%BD%93%E4%B8%8E%E8%83%8C%E6%99%AF%E9%A2%9C%E8%89%B2%E5%BC%82%E5%B8%B8)

| 操作系统                               | 全球                                   | 中国大陆                                                             |
|------------------------------------|--------------------------------------|------------------------------------------------------------------|
| Windows x86_64                     | [GitHub][github-win-x64]             | [主线][cf-win-x64] / [备线][ghproxy-win-x64]                         |
| macOS x86_64 (Intel 芯片)            | [GitHub][github-mac-x64]             | [主线][cf-mac-x64] / [备线][ghproxy-mac-x64]                         |
| macOS aarch64 (M 系列芯片)             | [GitHub][github-mac-aarch64]         | [主线][cf-mac-aarch64] / [备线][ghproxy-mac-aarch64]                 |
| Android universal (适用于所有设备)        | [GitHub][github-android]             | [主线][cf-android] / [备线][ghproxy-android]                         |
| Android arm64-v8a (仅新手机)           | [GitHub][github-android-arm64-v8a]   | [主线][cf-android-arm64-v8a] / [备线][ghproxy-android-arm64-v8a]     |
| Android armeabi-v7a (仅 32 位旧手机和电视) | [GitHub][github-android-armeabi-v7a] | [主线][cf-android-armeabi-v7a] / [备线][ghproxy-android-armeabi-v7a] |
| Android x86_64 (Chromebook 及模拟器)   | [GitHub][github-android-x86_64]      | [主线][cf-android-x86_64] / [备线][ghproxy-android-x86_64]           |

也可以扫描二维码下载 Android 版本 (适用于所有设备)：

[github-android-qr]: https://github.com/open-ani/ani/releases/download/v3.9.0-alpha02/ani-3.9.0-alpha02-universal.apk.github.qrcode.png

[cf-android-qr]: https://d.myani.org/v3.9.0-alpha02/ani-3.9.0-alpha02-universal.apk.cloudflare.qrcode.png

| 全球                           | 中国大陆                         |
|------------------------------|------------------------------|
| ![GitHub][github-android-qr] | ![Cloudflare][cf-android-qr] |

</details>

<hr /><em>This discussion was created from the release <a href='https://github.com/open-ani/ani/releases/tag/v3.9.0-alpha02'>3.9.0-alpha02</a>.</em>

**Comment by LiJoJun** [comment]
我遇到一个问题，不知道是普遍性的还是只是我个例，但总之还是反馈一下：我解压运行exe时，不论是双击还是使用管理员权限，等待许久都没有弹出窗口，然后我打开任务管理器，发现ani是有在运行的，然后我在任务管理中结束它的任务再重新打开，观察任务管理器，我发现与它一同结束和启动的还有powershell，然后我尝试单独结束powershell的进程，这时ani就可以正常弹出窗口了。

**Reply by LiJoJun** [reply]
> 我遇到一个问题，不知道是普遍性的还是只是我个例，但总之还是反馈一下：我解压运行exe时，不论是双击还是使用管理员权限，等待许久都没有弹出窗口，然后我打开任务管理器，发现ani是有在运行的，然后我在任务管理中结束它的任务再重新打开，观察任务管理器，我发现与它一同结束和启动的还有powershell，然后我尝试单独结束powershell的进程，这时ani就可以正常弹出窗口了。

我只是反映一下我遇到的现象，具体是什么问题我也不清楚，希望大牛解答

**Reply by Him188** [reply]
CC @hacbit

**Reply by hacbit** [reply]
> 我遇到一个问题，不知道是普遍性的还是只是我个例，但总之还是反馈一下：我解压运行exe时，不论是双击还是使用管理员权限，等待许久都没有弹出窗口，然后我打开任务管理器，发现ani是有在运行的，然后我在任务管理中结束它的任务再重新打开，观察任务管理器，我发现与它一同结束和启动的还有powershell，然后我尝试单独结束powershell的进程，这时ani就可以正常弹出窗口了。

能否复现呢，最好能发个视频给我看看。
ani进程在任务管理器可以看到，但是没有弹出窗口，我觉得不是ani或者ani的更新服务的问题。因为ani_update只负责启动进程，而不保证进程的具体行为。
然后你在任务管理器中结束ani进程，又重新打开。你是在结束ani前，更新服务的那个终端窗口是还没关闭吗？重新打开ani后，终端会跟着一起重启？！这个太奇怪了，ani在启动时应该是会不打开终端的，而如果终端本身如果已经被关闭的话也不可能自己重启。

**Comment by LiJoJun** [comment]
是的，我可以复现上述现象，并录频给你
[https://res.public.onecdn.static.microsoft/assets/mail/file-icon/png/generic_16x16.png]ani无法弹窗复现.rar<https://1drv.ms/u/c/afd90b53d7329b3f/EcLlkv5JFtlPppgpS0QiG3AB2EFPCc2rNaNlxzjVCE0Q0A>
________________________________

**Reply by hacbit** [reply]
> 是的，我可以复现上述现象，并录频给你 [[https://res.public.onecdn.static.microsoft/assets/mail/file-icon/png/generic_16x16.png]ani无法弹窗复现.rar<https://1drv.ms/u/c/afd90b53d7329b3f/EcLlkv5JFtlPppgpS0QiG3AB2EFPCc2rNaNlxzjVCE0Q0A](https://res.public.onecdn.static.microsoft/assets/mail/file-icon/png/generic_16x16.png]ani%E6%97%A0%E6%B3%95%E5%BC%B9%E7%AA%97%E5%A4%8D%E7%8E%B0.rar<https://1drv.ms/u/c/afd90b53d7329b3f/EcLlkv5JFtlPppgpS0QiG3AB2EFPCc2rNaNlxzjVCE0Q0A)>
> […](#)
> ________________________________ From: Hacbit ***@***.***> Sent: 09 September 2024 18:03 To: open-ani/ani ***@***.***> Cc: LiJoJun ***@***.***>; Comment ***@***.***> Subject: Re: [open-ani/ani] 3.9.0-alpha02 (Discussion #895) 我遇到一个问题，不知道是普遍性的还是只是我个例，但总之还是反馈一下：我解压运行exe时，不论是双击还是使用管理员权限，等待许久都没有弹出窗口，然后我打开任务管理器，发现ani是有在运行的，然后我在任务管理中结束它的任务再重新打开，观察任务管理器，我发现与它一同结束和启动的还有powershell，然后我尝试单独结束powershell的进程，这时ani就可以正常弹出窗口了。 能否复现呢，最好能发个视频给我看看。 ani进程在任务管理器可以看到，但是没有弹出窗口，我觉得不是ani或者ani的更新服务的问题。因为ani_update只负责启动进程，而不保证进程的具体行为。 然后你在任务管理器中结束ani进程，又重新打开。你是在结束ani前，更新服务的那个终端窗口是还没关闭吗？重新打开ani后，终端会跟着一起重启？！这个太奇怪了，ani在启动时应该是会不打开终端的，而如果终端本身如果已经被关闭的话也不可能自己重启。 ― Reply to this email directly, view it on GitHub<[#895 (reply in thread)](https://github.com/open-ani/ani/discussions/895#discussioncomment-10589168)>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AYN23DZANIQSOTIP6X3WMHDZVVW6DAVCNFSM6AAAAABNYGISZOVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTANJYHEYTMOA>. You are receiving this because you commented.Message ID: ***@***.***>

很疑惑我无法直接访问这个视频，你可以把这个视频直接用邮件发我（
